src/components/movieForm/MovieFormBase.js

```diff
@@ -0,0 +1,73 @@
+import React, {useState} from 'react';
+import {connect} from "react-redux";
+import Popup from "../system/Popup";
+import {cancelEditMovieAction} from "../../actions";
+import styles from '../../css/styles.module.css';
+
+const MovieFormBase = ({title, ...props}) => {
+    const [item, setItem] = useState({...props.source});
+
+    const reset = () => {
+        setItem({...props.source});
+    }
+
+    const onChangeHandler = (e) => {
+        item[e.target.name] = e.target.value;
```

Comment:
Code smells. What's going on here?

Reply:
Please don't manually mutate `item` to prevent bugs.
```javascript
const newItem = {...item, [e.target.name]: e.target.value};
setItem(newItem);
```